src/ansys/mechanical/core/embedding/logger/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""Embedding logger.
+
+Module to interact with the built-in logging system of Mechanical.
+
+Usage
+-----
+
+Configuring logger
+~~~~~~~~~~~~~~~~~~
+
+Configuring the logger can be done using the Configuration class, for instance:
+
+.. code:: python
+  import ansys.mechanical.core as mech
+  from ansys.mechanical.core.embedding.logger import Configuration, Logger
+
+  Configuration.configure(level=logging.INFO, to_stdout=True, base_directory=None)
+  app = mech.App(version=241)
+
+Then, the Logger class can be used to write messages to the log, for instance:
+
+.. code:: python
+
+   Logger.error("message")
+
+
+"""
+
+import logging
+import os
+import typing
+
+from ansys.mechanical.core.embedding import initializer
+from ansys.mechanical.core.embedding.logger import environ, linux_api, sinks, windows_api
+
+LOGGING_SINKS: typing.Set[int] = set()
+LOGGING_CONTEXT: str = "PYMECHANICAL"
+
+
+def _get_backend() -> (
+    typing.Union[windows_api.APIBackend, linux_api.APIBackend, environ.EnvironBackend]
+):
+    """Get the appropriate logger backend.
+
+    Before embedding is initialized, logging is configured via environment variables
+    After embedding is initialized, logging is configured by making API calls into the
+    mechanical logging system.
+   
+    However, the API is mostly the same in both cases, though some methods only work
+    in one of the two backends.
+    
+    Setting the base directory only works before initializing
+    Actually logging a message or flushing the log only works after initializing
+    """
+    # TODO - use abc instead of a union type?
+    embedding_initialized = initializer.INITIALIZED_VERSION != None
+    if not embedding_initialized:
+        return environ.EnvironBackend()
+    if os.name == "nt":
+        return windows_api.APIBackend()
+    return linux_api.APIBackend()
+
+
+class Configuration:
+    """Logger configuration for Mechanical embedding."""
```

Comment:
```suggestion
    """Configures logger for Mechanical embedding."""
```
A description for a class should start in a verb ending in an "s" or "es".